core/src/main/java/io/questdb/griffin/engine/functions/date/TimestampFloorFunctionFactory.java

```diff
@@ -40,28 +42,44 @@ public String getSignature() {
 
     @Override
     public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
-        final char c = args.getQuick(0).getChar(null);
+        final CharSequence str = args.getQuick(0).getStr(null);
+        char c = (str != null && str.length() == 1) ? str.charAt(0) : 0;
         switch (c) {
-            case 'd':
-                return new TimestampFloorFunctions.TimestampFloorDDFunction(args.getQuick(1));
             case 'M':
                 return new TimestampFloorFunctions.TimestampFloorMMFunction(args.getQuick(1));
             case 'y':
                 return new TimestampFloorFunctions.TimestampFloorYYYYFunction(args.getQuick(1));
-            case 'w':
-                return new TimestampFloorFunctions.TimestampFloorWWFunction(args.getQuick(1));
-            case 'h':
-                return new TimestampFloorFunctions.TimestampFloorHHFunction(args.getQuick(1));
-            case 'm':
-                return new TimestampFloorFunctions.TimestampFloorMIFunction(args.getQuick(1));
-            case 's':
-                return new TimestampFloorFunctions.TimestampFloorSSFunction(args.getQuick(1));
-            case 'T':
-                return new TimestampFloorFunctions.TimestampFloorMSFunction(args.getQuick(1));
-            case 0:
-                throw SqlException.position(argPositions.getQuick(0)).put("invalid kind 'null'");
             default:
-                throw SqlException.position(argPositions.getQuick(0)).put("invalid kind '").put(c).put('\'');
+                int stride = 1;
+                if (c == 0 && str != null && str.length() > 1) {
+                    c = str.charAt(str.length() - 1);
+                    try {
```

Comment:
If stride is malformed wouldn't it be better to throw SqlException and inform user like it's done later in code for bad unit ?